packages/server/dataloader/atlassianLoaders.ts

```diff
@@ -34,37 +33,53 @@ export const freshAtlassianAuth = (parent: RethinkDataLoader) => {
     async (keys) => {
       const results = await Promise.allSettled(
         keys.map(async ({userId, teamId}) => {
-          const atlassianAuth = await getAtlassianAuthByUserIdTeamId(userId, teamId)
-
-          if (!atlassianAuth?.refreshToken) {
+          const userAtlassianAuths = await getAtlassianAuthsByUserId(userId)
+          const atlassianAuthToRefresh = userAtlassianAuths.find(
+            (atlassianAuth) => atlassianAuth.teamId === teamId
+          )
+          if (!atlassianAuthToRefresh?.refreshToken) {
             // Not always an error! For suggested integrations, this won't exist.
             // sendToSentry(new Error('No atlassian access token exists for team member'), {
             //   userId,
             //   tags: {teamId}
             // })
             return null
           }
-          const {accessToken: existingAccessToken, refreshToken} = atlassianAuth
+          const {accessToken: existingAccessToken, refreshToken} = atlassianAuthToRefresh
           const decodedToken = existingAccessToken && (decode(existingAccessToken) as any)
           const now = new Date()
           const inAMinute = Math.floor((now.getTime() + 60000) / 1000)
           if (!decodedToken || decodedToken.exp < inAMinute) {
             const manager = await AtlassianServerManager.refresh(refreshToken)
             const {accessToken, refreshToken: newRefreshToken} = manager
-            atlassianAuth.accessToken = accessToken
-            atlassianAuth.updatedAt = now
-
-            if (newRefreshToken) {
-              atlassianAuth.refreshToken = newRefreshToken
-            }
+            const updatedRefreshToken = newRefreshToken ?? atlassianAuthToRefresh.refreshToken
+            // if user integrated the same Jira account with using different teams we need to update them as well
+            // reference: https://github.com/ParabolInc/parabol/issues/5601
+            const updatedSameJiraAccountAtlassianAuths = userAtlassianAuths
+              .filter(
+                (auth) =>
+                  auth.teamId !== teamId && auth.accountId === atlassianAuthToRefresh.accountId
+              )
+              .map((auth) => ({
+                ...auth,
+                accessToken,
+                refreshToken: updatedRefreshToken
+              }))
 
-            await upsertAtlassianAuth(atlassianAuth)
+            await upsertAtlassianAuths([
+              {
+                ...atlassianAuthToRefresh,
+                accessToken,
+                refreshToken: updatedRefreshToken
+              },
```

Comment:
this update logic seems to be the same as lines 63-67. what do you think of merging all atlassianAuths to update into one list and then mapping them all once to updatedAuths together? 

i think we should also remember to do set `updatedAt` field to now.

Reply:
@tiffanyhan it's set by the sql query itself

```
VALUES :auths ON CONFLICT ("userId", "teamId") DO
UPDATE
SET (
    "isActive",
    "updatedAt",
    "accessToken",
    "refreshToken",
    "cloudIds",
    "scope",
    "accountId",
    "teamId",
    "userId"
  ) = (
    TRUE,
    CURRENT_TIMESTAMP,
    EXCLUDED."accessToken",
    EXCLUDED."refreshToken",
    EXCLUDED."cloudIds",
    EXCLUDED."scope",
    EXCLUDED."accountId",
    EXCLUDED."teamId",
    EXCLUDED."userId"
  );
```

So I think we should be good, but please lmk if you see some advantages in setting that in the code 💡

Reply:
oh i missed that, no this looks good!